lib/open_project/translations/models/locales_updater.rb

```diff
@@ -0,0 +1,189 @@
+# todo those ought to be in the gemspec
+require 'tempfile'
+require 'crowdin-api'
+require 'yaml'
+require 'zip'
+require 'pathname'
+require 'fileutils'
+
+require_relative '../helpers/tmp_directory'
+require_relative './git_repository'
+
+class LocalesUpdater
+  extend TmpDirectory
+
+  def self.update_all_locales_of_all_repos(debug: true)
+    repos_to_update = plugins_with_locales
+
+    tmp_path = create_tmp_directory(delete: true)
+
+    Dir.chdir tmp_path do
+      repos_to_update.each do |name, specifics|
+        # todo each branch
+        uri = specifics[:uri]
+        set_crowdin_specifics(specifics)
+
+        FileUtils.rm_rf name
+        Dir.mkdir name
+
+        repo_path = name
+        git_repo = GitRepository.new(uri, repo_path)
+        git_repo.clone_or_pull
+
+        git_repo.checkout(branch)
+        # todo or should we merge this branch into the next ('push vs pull')
+        git_repo.merge(previous_branch, '-Xours') if previous_branch
+        git_repo.within_repo do
+          upload_english
+          request_build
+          download_and_replace_locales
+          fix_missing_keys
+        end
+        git_repo.add('config/locales')
+        git_repo.commit('update locales from crowdin')
+        git_repo.push(branch) unless debug
+      end
+    end
+    delete_tmp_folder(tmp_path) unless debug
+  end
+
+  def self.plugins_with_locales
+    # todo get list from file
+    # todo raise error if something is missing
+    # todo how to handle api_keys? i will commit this one for now and invalidate it later
+    {'OpenProject-Backlogs' =>
+     {
+       uri: 'git@github.com:finnlabs/openproject-backlogs',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.